src/nni_manager/training_service/reusable/channels/amlCommandChannel.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import * as component from "../../../common/component";
+import { delay } from "../../../common/utils";
+import { CommandChannel, RunnerConnection } from "../commandChannel";
+import { EnvironmentInformation, Channel } from "../environment";
+import { AMLEnvironmentInformation } from '../aml/amlConfig';
+import { EventEmitter } from 'events';
+import { AMLEnvironmentService } from "../environments/amlEnvironmentService";
+import { STDOUT } from "../../../core/commands";
+
+class AMLRunnerConnection extends RunnerConnection {
+}
+
+export class AMLCommandChannel extends CommandChannel {
+    private stopping: boolean = false;
+    private currentMessageIndex: number = -1;
+    // make sure no concurrent issue when sending commands.
+    private sendQueues: [EnvironmentInformation, string][] = [];
+    private metricEmitter: EventEmitter | undefined;
+    private readonly NNI_METRICS_PATTERN: string = `NNISDK_MEb'(?<metrics>.*?)'`;
+    
+    public constructor(commandEmitter: EventEmitter) {
+        super(commandEmitter);
+    }
+    public get channelName(): Channel {
+        return "aml";
+    }
+
+    public async config(_key: string, _value: any): Promise<void> {
+        switch (_key) {
+            case "MetricEmitter":
+                this.metricEmitter = _value as EventEmitter;
+                break;
+        }
+    }
+
+    public async start(): Promise<void> {
+        // start command loops
+        this.receiveLoop();
+        this.sendLoop();
```

Comment:
Do you want to start the two loops concurrently or sequentially?

Reply:
fixed, use Promise.all